pkg/cloudprovider/vsphere/loadbalancer/nsxt_broker.go

```diff
@@ -0,0 +1,219 @@
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package loadbalancer
+
+import (
+	"net/http"
+	"strings"
+
+	nsxt "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/loadbalancer"
+	"github.com/vmware/go-vmware-nsxt/manager"
+)
+
+// NsxtBroker is an internal interface to enable mocking the nsxt backend
+type NsxtBroker interface {
+	ListIPPools() (manager.IpPoolListResult, error)
+	ReadLoadBalancerService(id string) (loadbalancer.LbService, error)
+	CreateLoadBalancerService(service loadbalancer.LbService) (loadbalancer.LbService, error)
+	ListLoadBalancerServices() (loadbalancer.LbServiceListResult, error)
+	UpdateLoadBalancerService(service loadbalancer.LbService) (loadbalancer.LbService, error)
+	DeleteLoadBalancerService(id string) (statusCode int, err error)
+	CreateLoadBalancerVirtualServer(server loadbalancer.LbVirtualServer) (loadbalancer.LbVirtualServer, error)
+	ListLoadBalancerVirtualServers() (loadbalancer.LbVirtualServerListResult, error)
+	UpdateLoadBalancerVirtualServer(server loadbalancer.LbVirtualServer) (loadbalancer.LbVirtualServer, error)
+	DeleteLoadBalancerVirtualServer(id string) (statusCode int, err error)
+	CreateLoadBalancerPool(pool loadbalancer.LbPool) (loadbalancer.LbPool, error)
+	ReadLoadBalancerPool(id string) (loadbalancer.LbPool, error)
+	ListLoadBalancerPools() (loadbalancer.LbPoolListResult, error)
+	UpdateLoadBalancerPool(pool loadbalancer.LbPool) (loadbalancer.LbPool, error)
+	DeleteLoadBalancerPool(id string) (statusCode int, err error)
+	AllocateFromIPPool(ipPoolID string) (ipAddress string, statusCode int, err error)
+	ListIPPoolAllocations(ipPoolID string) (ipAddresses []string, statusCode int, err error)
+	ReleaseFromIPPool(ipPoolID, ipAddress string) (statusCode int, err error)
+
+	CreateLoadBalancerTCPMonitor(monitor loadbalancer.LbTcpMonitor) (loadbalancer.LbTcpMonitor, error)
+	ListLoadBalancerMonitors() (loadbalancer.LbMonitorListResult, error)
+	ReadLoadBalancerTCPMonitor(id string) (loadbalancer.LbTcpMonitor, error)
+	UpdateLoadBalancerTCPMonitor(monitor loadbalancer.LbTcpMonitor) (loadbalancer.LbTcpMonitor, error)
+	DeleteLoadBalancerMonitor(id string) (int, error)
+}
+
+type nsxtBroker struct {
+	client *nsxt.APIClient
+}
+
+// NewNsxtBroker creates a new NsxtBroker to the real API
+func NewNsxtBroker(client *nsxt.APIClient) NsxtBroker {
+	return &nsxtBroker{client: client}
+}
+
+func (b *nsxtBroker) ListIPPools() (manager.IpPoolListResult, error) {
+	result, _, err := b.client.PoolManagementApi.ListIpPools(b.client.Context, nil)
```

Comment:
seems like this is still on Manager API, which is being deprecated. might be better if we switch to the Policy API.

Reply:
Do you have a code snipped with a better coding?

Reply:
We've switched to the new SDK